st2client/st2client/models/core.py

```diff
@@ -524,4 +524,10 @@ def listen(self, events=None, **kwargs):
         url = url + query_string
 
         for message in SSEClient(url):
+
+            # If the execution on the API server takes too long, the message
+            # can be empty. In this case, rerun the query.
+            if not message.data:
```

Comment:
Good catch :+1: 

And yeah we periodically send empty message / line break on hear beat to keep the connection open.